File: C/projects/DALEC_CODE/DALEC_ALL/HYDROLOGY_MODULES/CONVERTERS/HYDROFUN_MOI2CON.c
#pragma once

/*
Convert volumetric soil moisture (m3/m3) to conductivity (m/s)
Inputs:
sm - Soil moisture (m3/m3)
k0 - saturated hydraulic conductivity (m/s)
b - retention parameter
*/

double HYDROFUN_MOI2CON(double sm, double k0, double b){

double moi=k0*pow(sm,2*b+3);
 
return moi;
}

File: PYTHON/projects/CARDAMOM_MODELS/DALEC/DALEC_ALL/HYDROLOGY_MODULES/CONVERTERS/HYDROFUN_MOI2CON.py
import jax.numpy as jnp

def HYDROFUN_MOI2CON(moi, k0, b):
    """Convert volumetric soil moisture (m3/m3) to conductivity (m/s)"""
    return k0 * jnp.power(jnp.maximum(moi, 1e-9), 2.0 * b + 3.0)
